contract/src/lib.rs

```diff
@@ -548,38 +628,45 @@ impl BtcLightClient {
             return;
         }
 
-        let interval_tail_header_hash = self
-            .mainchain_height_to_header
-            .get(&(prev_block_header.block_height + 1 - config.blocks_per_adjustment))
-            .unwrap_or_else(|| env::panic_str(ERR_KEY_NOT_EXIST));
+        #[cfg(not(feature = "dogecoin"))]
+        let first_block_height = prev_block_header.block_height + 1 - config.blocks_per_adjustment;
+
+        #[cfg(feature = "dogecoin")]
```

Comment:
We should avoid using a feature flags a lot in the middle of the code, let's move the Dogcoin implementation to seperate file and add the link to code reference

Reply:
fixed: https://github.com/Near-One/btc-light-client-contract/pull/101/commits/3944df71a7eaaee6ca614bb42104c7223c858cbd